ci-operator/step-registry/idp/freeipa/idp-freeipa-commands.sh

```diff
@@ -0,0 +1,403 @@
+    temp_dir=$(mktemp -d)
+    openssl genrsa -out "$temp_dir"/ca.key 2048
+    openssl req -key "$temp_dir"/ca.key -nodes -new -out "$temp_dir"/ca.csr -subj '/CN=freeipa.freeipa.freeipa.svc.cluster.local/O=FREEIPA'
+    openssl req -x509 -new -key "$temp_dir"/ca.key -nodes -out "$temp_dir"/ca.crt -subj '/CN=freeipa.freeipa.freeipa.svc.cluster.local/O=FREEIPA' -days 365 -set_serial 1
+
+    oc create secret tls freeipa-certs --cert="$temp_dir"/ca.crt --key="$temp_dir"/ca.key -n freeipa
+
+    #Create a ConfigMap inside the openshift-config namespace to be used by the identity provider.
+    oc create configmap ipa-ca --from-file=ca.crt="$temp_dir"/ca.crt -n openshift-config
+
+    # Create a temporary file with a .yaml extension
+    temp_file=$(mktemp /tmp/freeipa_manifest.XXXXXX.yaml)
+    # Add the manifest to the temporary file
+    cat <<'EOF' > "$temp_file"
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: freeipa
+  name: freeipa
+  namespace: freeipa
+spec:
+  replicas: 2
+  selector:
+    matchLabels:
+      app: freeipa
+  template:
+    metadata:
+      labels:
+        app: freeipa
+    spec:
+      hostname: freeipa
+      subdomain: freeipa
+      initContainers:
+      - name: freeipa-openshift-container-1-step
+        image: quay.io/freeipa/freeipa-server:fedora-39
+        securityContext:
+          runAsUser: 0
+        tty: true
+        env:
+        - name: DEBUG_TRACE
+          value: ""
+        - name: IPA_SERVER_INSTALL_OPTS
+          value: "--unattended --external-ca --realm=IPA.EXAMPLE.COM  --domain=ipa.example.com --no-host-dns"
+        - name: KUBERNETES
+          value: "1"
+        - name: IPA_ADMIN_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: freeipa
+              key: IPA_ADMIN_PASSWORD
+        - name: PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: freeipa
+              key: IPA_DM_PASSWORD
+        - name: SYSTEMD_OFFLINE
+          value: "1"
+        - name: SYSTEMD_NSPAWN_API_VFS_WRITABLE
+          value: "network"
+        volumeMounts:
+          - name: data
+            mountPath: /data
+          - name: systemd-tmp
+            mountPath: /tmp
+          - name: systemd-var-run
+            mountPath: /var/run
+          - name: systemd-var-dirsrv
+            mountPath: /var/run/dirsrv
+      - name: setup-nssdb
+        image: quay.io/freeipa/freeipa-server:fedora-39
+        securityContext:
+          runAsUser: 0
+        command: ["/bin/bash", "-c"]
+        args:
+          - |
+            set -e
+            set -x
+            DATA="/data"
+            NSSDB=/tmp/nssdb-$RANDOM
+            PASSWORD_FILE=/tmp/nssdb-$RANDOM
+            echo $RANDOM > $PASSWORD_FILE
+            openssl pkcs12 -export -in /certs/tls.crt -inkey /certs/tls.key -out "$DATA"/ca.p12 -name "/CN=freeipa.freeipa.freeipa.svc.cluster.local/O=FREEIPA" -passout file:$PASSWORD_FILE
```

Comment:
`man openssl-pkcs12` shows `-name friendlyname: specifies the "friendly name" for the certificates and private key`. So you can give it a "friendly name":
```suggestion
            openssl pkcs12 -export -in /certs/tls.crt -inkey /certs/tls.key -out "$DATA"/ca.p12 -name "External CA" -passout file:$PASSWORD_FILE
```

Reply:
Updated, Thanks